en/lessons/parsers/introduction.md

```diff
@@ -0,0 +1,62 @@
+---
+version: 1.0.0
+title: Introduction to Parser Combinators
+---
+
+{% include toc.html %}
+
+Parsers are ubiquitous in Haskell litterature because, thanks to Parser Combinators it illustrates the power of composition.
+
+Parsers, while being mostly associated to compilers and interpreters, can be found everywhere.
+
+Their goal is to take a input data (usually a text or a stream), and produce a structured output (usually a data structure, such as an [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree)), see [this article](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/) for day to day application.
+
+Parser Combinators are a way to build parsers by composition.
+
+Let's say we want to extract all the tuples of a text, we can start by defining our tuple parser:
+
+```haskell
+import Text.ParserCombinators.ReadP
+
+tuple :: ReadP (String, String)
+tuple =
+  between (char '(') (char ')') $ do
+    left <- munch1 (/= ',')
+    char ','
+    skipSpaces
+    right <- munch1 (/= ')')
+    return (left, right)
+```
+
+As you can see, relying on `Monad`, allows us to focus on the structure on the input, instead of the structure on the Parser.
+
+then, we can define our main parser, composition:
+
+```haskell
+tuples :: ReadP [(String, String)]
+tuples = many tuple
+```
+
+that's the power of Parser Combinator, since `ReadP` implements `Alternative`, any parser can be reused and extended (here via [`many`](https://hackage.haskell.org/package/base/docs/Control-Applicative.html#v:many)).
```

Comment:
```suggestion
This is the power of parser combinators: since `ReadP` implements `Alternative`, any parser can be reused and extended (here via [`many`](https://hackage.haskell.org/package/base/docs/Control-Applicative.html#v:many)).
```
"This" may sound more natural here: "this" refers to things that have been mentioned very recently. 

The change in punctuation was suggested to remove a comma splice.